### PR TITLE
[OAP-2057][oap-native-sql]package dependencies into the jar file

### LINKED ISSUE #2057: [oap-native-sql]Package all dependency libraries into NativeSQL jar
The goal is to run the native SQL jar standalone without any dependencies issue.
There are 4 dependencies that must be solved in the current release:
1. libarrow.so
2. libgandiva.so
3. libparquet.so
4. libprotobuf.so

### LINKED ISSUE #2058: [oap-native-sql]Add arrow header files in the jar file
Since some arrow header file would be used in other libraries, we have to add most arrow header files into the jar file.
The header files should be placed into third_party/arrow directory.

### PR DESCRIPTION
**Pull request by weiting-chen**
## What changes were proposed in this pull request?

1. Add Arrow Header Files into third_party/arrow
2. Add libarrow.so.17 into the jar file
3. Add libgandiva.so.17 into the jar file
4. Add libparquet.so.17 into the jar file
5. Set to build libprotobuf from source to true

Closes #2057 
Closes #2058 

(Please fill in changes proposed in this fix)

## How was this patch tested?

Tested on my own development server and another server
(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by zhouyuan** [comment]
the travis failures seems due to some unit tests failing which may be unrelated. i'm ok if this passed local TPC-H/TPC-DS tests.

File: oap-native-sql/cpp/src/third_party/arrow/api.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```
Comment (zhouyuan) [inline oap-native-sql/cpp/src/third_party/arrow/api.h:1]: let's package all arrow headers

Comment (weiting-chen) [reply oap-native-sql/cpp/src/third_party/arrow/api.h:1]: Done to copy all arrow headers into third_party/arrow.

**Comment by weiting-chen** [comment]
> the travis failures seems due to some unit tests failing which may be unrelated. i'm ok if this passed local TPC-H/TPC-DS tests.

Done to pass all the TPC-H queries.
Some queries cannot complete in TPC-DS, checking if this is from some features enabling.
However, assume those fails wouldn't be related to a dependency related and a package issue, since most of the queries can run pass with the jar we packaged.

**Comment by zhouyuan** [comment]
@weiting-chen can you please pr this to the new repo? 
https://github.com/oap-project/native-sql-engine

**Comment by weiting-chen** [comment]
> @weiting-chen can you please pr this to the new repo?
> https://github.com/oap-project/native-sql-engine

Sure, np.

**Pull request merged**